enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js

```diff
@@ -772,5 +775,131 @@ describeWithToken("formatting > sandboxes", () => {
+            },
+            card_id: CARD_ID,
+            group_id: COLLECTION_GROUP_ID,
+            table_id: ORDERS_ID,
+          });
+        });
+        cy.log("**-- 2. Create the second native question with a filter --**");
+
+        cy.request("POST", "/api/card", {
+          name: "EE_520_Q2",
+          dataset_query: {
+            database: 1,
+            native: {
+              query:
+                "SELECT * FROM PRODUCTS↵WHERE CATEGORY={{sandbox}} AND PRICE>10",
+              "template-tags": {
+                sandbox: {
+                  "display-name": "Sandbox",
+                  id: "3d69ba99-7076-2252-30bd-0bb8810ba895",
+                  name: "sandbox",
+                  type: "text",
+                },
+              },
+            },
+            type: "native",
+          },
+          display: "table",
+          visualization_settings: {},
+        }).then(({ body: { id: CARD_ID } }) => {
+          cy.log(
+            "**-- 2a. Sandbox `Products` table based on this question --**",
+          );
+
+          cy.request("POST", "/api/mt/gtap", {
+            attribute_remappings: {
+              [ATTR_CAT]: ["variable", ["template-tag", "sandbox"]],
+            },
+            card_id: CARD_ID,
+            group_id: COLLECTION_GROUP_ID,
+            table_id: PRODUCTS_ID,
+          });
+        });
+
+        cy.log("**-- 3. Fetch permissions graph --**");
+
+        cy.request("GET", "/api/permissions/graph", {}).then(
+          ({ body: { groups, revision } }) => {
+            // Update permissions for `collections` group [id: 4]
+            // This mutates the original `groups` object => we'll pass it next to the `PUT` request
+            groups[COLLECTION_GROUP_ID] = {
+              1: {
+                schemas: {
+                  PUBLIC: {
+                    [PRODUCTS_ID]: { query: "segmented", read: "all" },
+                    [ORDERS_ID]: { query: "segmented", read: "all" },
+                  },
+                },
+              },
+            };
+
+            cy.log("**-- 4. Update/save permissions --**");
+
+            cy.request("PUT", "/api/permissions/graph", {
+              groups,
+              revision,
+            });
+          },
+        );
+      });
+
+      signOut();
+
+      cy.log("**-- Logging in as sandboxed user --**");
+      cy.request("POST", "/api/session", {
+        username: sandboxed_user.email,
+        password: sandboxed_user.password,
+      });
+
+      openOrdersTable();
+
+      // View details for Product ID #14
+      cy.get(".cellData")
+        .contains("14")
+        .click();
+      cy.findByText(/View details/i).click();
+
+      cy.log("**-- Reported failing on v1.36.x --**");
+      cy.log(
+        "**It should show object details instead of filtering by this Product ID**",
+      );
+      // The name of this Vendor is visible in "details" only
+      cy.findByText("McClure-Lockman");
```

Comment:
There should be 2 assertions for this test.
1. It should show remapping correctly on FK (this is the default on setups, I don't know if it is different on test data, otherwise it needs to be enabled for Admin > Data Model > Sample Dataset > Orders > Product ID :gear: > Display values = Use foreign key. It should only be enabled for Product ID, if it is also enabled for User ID, then it will likely cause a permission error)
2. And be able to click into Object Details from a FK

```suggestion
      cy.log("**-- Reported failing on v1.36.x --**");
      cy.log(
        "**It should show remapped Display Values instead of Product ID**",
      );
      // View details for Product ID #14 remapped
      cy.get(".cellData")
        .contains("Awesome Concrete Shoes")
        .click();
      cy.findByText(/View details/i).click();

      cy.log(
        "**It should show object details instead of filtering by this Product ID**",
      );
      // The name of this Vendor is visible in "details" only
      cy.findByText("McClure-Lockman");
```

Reply:
Oh, that's why I had the wrong assertion first time around thinking that product title/name is visible only in details page. No, it's definitely not enabled for testing data. I'll have to add that to the test.

Reply:
Ok, added remapping at the very beginning of this test and adjusted the assertion. Everything should be covered now. Phew...